src/main/java/com/meilisearch/sdk/Index.java

```diff
@@ -137,4 +138,45 @@ public String getUpdate(int updateId) throws Exception {
 		Gson gson = new Gson();
 		return gson.fromJson(this.documents.getUpdates(this.uid), UpdateStatus[].class);
 	}
+
+	/**
+	 * Wait for a pending update to be processed
+	 *
+	 * @param updateId ID of the index update
+	 * @param timeoutInMs number of milliseconds before throwing and Exception
+	 * @param intervalInMs number of milliseconds before requesting the status again
+	 * @throws Exception If timeout is reached
+	 */
+	public void waitForPendingUpdate(int updateId) throws Exception {
+		waitForPendingUpdate(updateId, 5000, 50);
+	}
+
+	/**
+	 * Wait for a pending update to be processed
+	 *
+	 * @param updateId ID of the index update
+	 * @param timeoutInMs number of milliseconds before throwing an Exception
+	 * @param intervalInMs number of milliseconds before requesting the status again
+	 * @throws Exception if timeout is reached
+	 */
+	public void waitForPendingUpdate(int updateId, int timeoutInMs, int intervalInMs) throws Exception {
```

Comment:
maybe something like a `CompletableFuture<UpdateStatus>` would be better for something like this.
Supply the updateId -> thenApply(...) the while loop -> call .get(timeout)